service/src/main/java/uk/gov/hmcts/reform/fpl/service/legalcounsel/ManageLegalCounselService.java

```diff
@@ -0,0 +1,169 @@
+            .orElse(emptyList());
+    }
+
+    public void updateLegalCounsel(CaseDetails caseDetails) {
+        String caseId = caseDetails.getId().toString();
+        List<SolicitorRole> caseSolicitorRoles = caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(caseId);
+
+        CaseData caseData = caseConverter.convert(caseDetails);
+        List<Element<LegalCounsellor>> legalCounsellors =
+            caseData.getManageLegalCounselEventData().getLegalCounsellors();
+
+        Map<String, Object> data = caseDetails.getData();
+
+        List<Element<Respondent>> allRespondents = caseData.getAllRespondents();
+        caseSolicitorRoles.stream()
+            .filter(solicitorRole -> solicitorRole.getRepresenting() == RESPONDENT)
+            .map(SolicitorRole::getIndex)
+            .map(allRespondents::get)
+            .map(Element::getValue)
+            .forEach(respondent -> respondent.setLegalCounsellors(legalCounsellors));
+        data.put("respondents1", allRespondents);
+
+        List<Element<Child>> allChildren = caseData.getAllChildren();
+        caseSolicitorRoles.stream()
+            .filter(solicitorRole -> solicitorRole.getRepresenting() == CHILD)
+            .map(SolicitorRole::getIndex)
+            .map(allChildren::get)
+            .map(Element::getValue)
+            .forEach(child -> child.setLegalCounsellors(legalCounsellors));
+        data.put("children1", allChildren);
+
+        data.remove("legalCounsellors");
+    }
+
+    public List<String> validateEventData(CaseDetails caseDetails) {
+        List<String> errorMessages = new ArrayList<>();
+
+        CaseData caseData = caseConverter.convert(caseDetails);
+        ManageLegalCounselEventData manageLegalCounselEventData = caseData.getManageLegalCounselEventData();
+        List<Element<LegalCounsellor>> legalCounsellors = manageLegalCounselEventData.getLegalCounsellors();
+
+        legalCounsellors.stream()
+            .map(Element::getValue)
+            .filter(legalCounsellor -> organisationService.findUserByEmail(legalCounsellor.getEmail()).isEmpty())
+            .map(legalCounsellor -> format(UNREGISTERED_USER_ERROR_MESSAGE_TEMPLATE, legalCounsellor.getFullName()))
+            .forEach(errorMessages::add);
+
+        legalCounsellors.stream()
+            .map(Element::getValue)
+            .filter(legalCounsellor -> Optional.ofNullable(legalCounsellor.getOrganisation())
+                .map(uk.gov.hmcts.reform.ccd.model.Organisation::getOrganisationID)
+                .filter(not(Strings::isNullOrEmpty))
+                .isEmpty())
+            .map(legalCounsellor -> format("Legal counsellor %s %s has no selected organisation",
+                legalCounsellor.getFirstName(),
+                legalCounsellor.getLastName()))
+            .forEach(errorMessages::add);
+
+        return errorMessages;
+    }
+
+    public void runFinalEventActions(CaseDetails previousCaseDetails, CaseDetails currentCaseDetails) {
```

Comment:
vv minor but current convention would be to return the list of events and then have the controller publish them

Reply:
Done